hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -248,26 +271,6 @@ private void addEncryptionKeyRequestHeaders(String path,
         // get new context for create file request
         requestHeaders.add(new AbfsHttpHeader(X_MS_ENCRYPTION_CONTEXT,
             encryptionAdapter.getEncodedContext()));
-      } else if (encryptionAdapter == null) {
-        // get encryption context from GetPathStatus response header
-        byte[] encryptionContext;
-        try {
-          encryptionContext = getPathStatus(path, false, tracingContext)
-              .getResult().getResponseHeader(X_MS_ENCRYPTION_CONTEXT)
-              .getBytes(StandardCharsets.UTF_8);
-        } catch (NullPointerException e) {
-          LOG.debug("EncryptionContext not present in GetPathStatus response.");
-          throw new IOException(
-              "EncryptionContext not present in GetPathStatus response", e);
-        }
-        try {
-          encryptionAdapter = new EncryptionAdapter(encryptionContextProvider,
-              new Path(path).toUri().getPath(), encryptionContext);
-          encryptionAdapterCreated = true;
-        } catch (IOException e) {
-          LOG.debug("Could not initialize EncryptionAdapter");
-          throw e;
-        }
```

Comment:
Inline with https://github.com/apache/hadoop/pull/3440#pullrequestreview-1133031625. AbfsClient to not have heuristic to create encryptionAdapter.